src/parser/lval.js

```diff
@@ -378,4 +380,18 @@ export default class LValParser extends NodeUtils {
       }
     }
   }
+
+  checkToRestConversion(node: SpreadElement): void {
+    const validArgumentTypes = [
```

Comment:
out of curiosity, does having rest's argument being an array pattern inside an array pattern have any purpose? besides that its possible to do ;p

it seems those are equivalents:
```js
// 1.
var [a, ...[b, ...c]] = [1,2,4,5,6]
```
```js
// 2.
var [a, b, ...c] = [1,2,4,5,6]
```
It seems to me that this is the same case which got now forbidden for object's rest:
```js
 ( {...{a, b}} = {} ) 
```
which simply didnt do much (to my understanding), just unwrapping/flattening this nested pattern